internal/v1/server.go

```diff
@@ -870,3 +848,22 @@ func (s *Server) HTTPErrorHandler(err error, c echo.Context) {
 		}
 	}
 }
+
+func noAssociateAccounts(nextHandler echo.HandlerFunc) echo.HandlerFunc {
+	return func(ctx echo.Context) error {
+		idh, err := getIdentityHeader(ctx)
+		if err != nil {
+			return err
+		}
+
+		if idh.Identity.Type == "Associate" {
+			// Associate account types are not guaranteed to have an associated org_id, these accounts
+			// should not be able to access image-builder as long as we don't explicitly enable turnpike
+			// access, or another such service forwards them to us. Explicitly reject such accounts for
+			// now.
+			return echo.NewHTTPError(http.StatusBadRequest, "unsupported account type: 'Associate'")
+		}
```

Comment:
So I wonder if we can't just introduce a dummy org_id value here for associates?  Like if suddenly a bunch of internal people lose access, that could be bad no?

Reply:
This would require explicit support beyond just the org_id. These accounts are not used to access image builder today (basically: not what it looks like). Happy to discuss though.

Reply:
This would require explicit support beyond just the org_id. These accounts are not used to access image builder today (basically: not what it looks like). Happy to discuss though.